### ISSUE TITLE
Why is the preprocessing only done in files with capital file extensions?

### ISSUE THREAD

**Issue by PaulOxxx1** [hansec/fortran-language-server#126]
Hello again! 😁 

I'm just wondering why the language server somewhat arbitrarily only preprocesses the files with capital letters in the file extension ... and if there is a way to enable the preprocessing for all source files. Perhaps there could be an option in the ".fortls" file that I can set so that every file is preprocessed?

The problem is that I don't want to rename all file extensions in my project, because that would probably be a pain of a git commit, since it would basically modify every single file and we wouldn't be able to easily merge work from older branches which my colleagues are still using. Also it just seems a bit irrational in general, and I'd probably be judged by everyone in the office because I need my file extensions to be capital letters... 😅 

Also I'm still interested in issue #108 ... but I understand that it's a bit exotic.

**Comment by PaulOxxx1** [comment]
I just realized that you've already been working on this in a separate branch. Anyways thank you!

**Comment by hansec** [comment]
The reason for this is that several common Fortran compilers (gcc, intel) only do preprocessing on capitalized extensions by default (ie without a flag). Support for specifying file extensions for preprocessing should now be available in `v1.10`. If you notice any issues with this or it doesn't cover all your use cases let me know.

**Comment by PaulOxxx1** [comment]
It works pretty nicely for my use case, thanks a lot! 

I added the .h suffix to the array of recognised suffixes, and I noticed that the .h files I (unfortunately have to) deal with now also get parsed if I open them once in visual studio code. (See issue #108). But this means I would have to open all my .h files once every time I reopen my editor.

I‘m not sure if this behaviour is intended ... probably not ... but for me it’s a nice side effect and _almost_ what I want! :)
